packages/rx_bloc_cli/mason_templates/bricks/feature_profile/__brick__/lib/feature_profile/blocs/profile_bloc.dart

```diff
@@ -46,37 +43,18 @@ class ProfileBloc extends $ProfileBloc {
   Stream<bool> _mapToIsLoadingState() => loadingState;
 
   @override
-  ConnectableStream<Result<bool>> _mapToSyncNotificationsStatusState() =>
+  ConnectableStream<Result<bool>> _mapToAreNotificationsEnabledState() =>
       Rx.merge([
-        _$setNotificationsEvent.switchMap(
-          (subscribePushNotifications) {
-            if (subscribePushNotifications) {
-              return _notificationService
-                  .subscribe()
-                  .then((_) => true)
-                  .asResultStream(tag: tagNotificationSubscribe);
-            }
-
-            return _notificationService
-                .unsubscribe()
-                .then((_) => false)
-                .asResultStream(tag: tagNotificationUnsubscribe);
-          },
-        ),
+        _$toggleNotificationsEvent.switchMap((_) => _notificationService
+            .toggleNotifications()
+            .asStream()
+            .asResultStream()),
         _$loadNotificationsSettingsEvent.startWith(null).switchMap((value) =>
             _notificationService
                 .syncNotificationSettings()
-                .then((_) => true)
+                .asStream()
+                .switchMap((_) =>
+                    _notificationService.areNotificationsEnabled().asStream())
```

Comment:
Sync notification and are enabled methods can be grouped in a single private function to reduce the used switch map operators

Reply:
Done